### PR TITLE
Client polish

### PR DESCRIPTION
**Pull request by Esterni**
Cleaned up different areas of the client module. Re-named variables for consistency and clarity. Fixed some default parameters. 

Might be easier to view commit by commit on this one

### CODE REVIEW

File: pyracing/client.py [hunk 1]
```diff
@@ -48,8 +42,9 @@ def __init__(self, username: str, password: str, log=default_logger()):
         self.password = password
         self.session = httpx.AsyncClient()
         self.log = log
+        self.__initial_auth = False
 
-    async def authenticate(self):
+    async def _authenticate(self):
```
Comment (Esterni) [inline pyracing/client.py:47]: while this makes it private, it does not prevent it from being accessed. Private classes in Python are not made inaccessible, it's just a suggestion.

File: pyracing/client.py [hunk 2]
```diff
@@ -73,13 +68,16 @@ async def authenticate(self):
                 'visiting members.iracing.com'
             )
         else:
+            self.initial_auth = True
             self.log.info('Login successful')
 
-    # Wrapper for all functions that builds the final self.session.get()
-
-    async def __build_request(self, url, params):
+    async def _build_request(self, url, params):
```
Comment (Esterni) [inline pyracing/client.py:74]: Learned that two underscores is for name mangling, one underscore is for private methods

File: pyracing/constants.py [hunk 3]
```diff
@@ -245,7 +247,7 @@ def number_to_string(self, number):
             return None
 
 
-class SimSesNum:
+class SimSesId:
```
Comment (XanderRiga) [inline pyracing/constants.py:250]: this should maybe be `SimSessionType` or something? It really refers to a type, not an id

File: pyracing/constants.py [hunk 4]
```diff
@@ -144,6 +142,10 @@
 URL_MY_RACERS = (mSite + '/GetDriverStatus')
 URL_MEM_DIVISION = (mSite + '/GetMembersDivision')
 
+# iRacing rounds down to the 5 minute mark - DO NOT CHANGE
+class Date:
```
Comment (XanderRiga) [inline pyracing/constants.py:146]: This should just be a single method, no need for a class here

File: pyracing/client.py [hunk 5]
```diff
@@ -195,7 +191,7 @@ async def current_seasons(
         field_dict = {
             'year': year,
             'quarter': quarter,
-            'seriesshortname': series_short_name,
+            'seriesshortname': series_name_short,
```
[the remark this answers is not in the document]
Comment (Esterni) [reply pyracing/client.py:194]: If it is, that's a mistake. I tried to follow the `subject_noun_adjective` structure. So `name` would be the noun of the subject `series` and `short` is the descriptor

File: pyracing/constants.py (hunk 3 above)
Comment (Esterni) [reply pyracing/constants.py:250]: Haha yeah, I've been going back and forth on this, trying to find a good description. SimSessionType sounds good.

File: pyracing/constants.py (hunk 4 above)
Comment (Esterni) [reply pyracing/constants.py:146]: I made it a class to possibly add more options. An example would be the search queries for team_standings which offer a time lowerbound and time upperbound

**Pull request merged**